pkg/registry/core/service/rest.go

```diff
@@ -570,3 +558,82 @@ func (rs *REST) allocateHealthCheckNodePort(service *api.Service) error {
 	}
 	return nil
 }
+
+// The return bool value indicates if the caller should release clusterIP before return
+func (rs *REST) initClusterIP(service *api.Service) (bool, error) {
```

Comment:
Please add an unit test for `initClusterIP` and `updateNodePort`. Let me know if this is too much trouble to fake interfaces. I think it should be doable for these two.